src/Microsoft.AspNetCore.DataProtection/DataProtectionUtilityExtensions.cs

```diff
@@ -37,10 +36,6 @@ public static string GetApplicationUniqueIdentifier(this IServiceProvider servic
             if (services != null)
             {
                 discriminator = services.GetService<IApplicationDiscriminator>()?.Discriminator;
```

Comment:
@davidfowl not really a good idea. Even if this service is now registered by the hosting block, there are cases where no `IApplicationDiscriminator` will be available: when using DP in a non-ASP.NET Core app or in an OWIN/Katana app using the interop package.